p4-16/spec/P4-16-spec.adoc

```diff
@@ -5498,6 +5498,12 @@ types (e.g. ``struct``s) are copied recursively, and all components
 of ``header``s are copied, including "validity" bits. Assignment is
 not defined for `extern` values.
 
+An assignment may also be written with a binary operator immediately
+before the `=` sign.  This performs the binary operator on the old
```

Comment:
This only works for selected binary operations and that should be clearly stated. While you have included all arithmetic operators as far as I can see, there are more binary operators that are not included:

- the boolean operators `&&` and `||` (note that this relates also to https://github.com/p4lang/p4c/pull/5108#discussion_r1936243149)
- the relational operators, which are also binary operators, but they return boolean, so it does not make sense (neither from type point not from usability point) to use them in compound operators.
- the `++` string operator (which needs not be included as there are no string variables, and even if there were they would still be `const`)
- the mask (`&&&`) and range (`..`) operators, again, these don't make sense here.